numpy/ma/mrecords.py

```diff
@@ -376,7 +375,6 @@ def view(self, dtype=None, type=None):
             try:
                 if issubclass(dtype, ndarray):
                     output = ndarray.view(self, dtype)
-                    dtype = None
                 else:
                     output = ndarray.view(self, dtype)
```

Comment:
It looks like this entire `if`/`else` statement is in fact redundant

Reply:
looks like `output` is indeed being returned at the end. 
why would you say `if`/`else` is redundant ? since its actually being put to use, can we discuss this further before i can make a commit ?
Thanks.

Reply:
hey, I did what you suggested to `mrecords.py` and ran its test, It failed. The code passes all the tests if I revert the change tho.
Here are the errors - 
```
TypeError: Cannot cast scalar from dtype((numpy.record, [('a', '<f8'), ('b', '<f8')])) to dtype('float64') according to the rule 'unsafe'

Reply:
Ugh, at this point I'd just keep the `if` / `else` statement in.
The code is a bit wonky IMO, but the sole purpose of the `issubclass` check is trigger an exception, thus signaling that `dtype: npt.DTypeLike` has been used rather than `dtype: Type[np.ndarray]`.
https://github.com/numpy/numpy/blob/12c6000601c7665e52502ab4a67a54d290498266/numpy/ma/mrecords.py#L375-L394

Reply:
sure, restored the change. New commit up now. 
Thanks